src/redux/slices/addressBook.slice.ts

```diff
@@ -5,17 +5,29 @@ import {
 import { Network } from '../networks';
 import {RootState} from "../store";
 import {REHYDRATE} from "redux-persist";
+import { ensApi } from './ensResolver.slice';
+import { add } from 'lodash';
 
 export interface AddressBookEntry {
   chainId: number,
   address: string,
-  nameTag: string
+  nameTag: string,
+  ensName: string
 }
 
-export const createEntryId = (network: Network, address: string) => (`${network.chainId}_${address.toLowerCase()}`);
+export const createEnsName = (address: string) => {
+  if (address !== null) {
+    const ensQuery = ensApi.useResolveNameQuery(
+      address
+    )
+    return ensQuery.name
+    }
+}
+
+export const createEntryId = (network: Network, address: string) => (`${network.chainId}_${address.toLowerCase()}_${createEnsName(address)}`);
 
 export const getEntryId = (addressBookEntry: AddressBookEntry) => {
-  return `${addressBookEntry.chainId}_${addressBookEntry.address.toLowerCase()}`;
+  return `${addressBookEntry.chainId}_${addressBookEntry.address.toLowerCase()}_${addressBookEntry.ensName}`;
```

Comment:
I'm not sure we want to put anything ENS into the address book... because ENS lookups should be on-demand. Let me think about this a little bit.